src/rt/lifetime.d

```diff
@@ -1823,15 +1819,7 @@ byte[] _d_arrayappendcTX(const TypeInfo ti, ref byte[] px, size_t n)
                 }
 
                 // couldn't do it, reallocate
-                info = GC.qalloc(newcap + LARGEPAD, info.attr);
-                __setArrayAllocLength(info, newsize, isshared);
-                if(!isshared)
-                    __insertBlkInfoCache(info, bic);
-                auto newdata = cast(byte *)info.base + LARGEPREFIX;
-                memcpy(newdata, px.ptr, length * sizeelem);
-                // do postblit processing
-                __doPostblit(newdata, length * sizeelem, ti.next);
-                (cast(void **)(&px))[1] = newdata;
```

Comment:
OK, found it.
The code slightly changed the stack layout, thereby getting rid of a false pointer that pinned one of the big words array.
Therefor the old code grew the heap (used an additional pool) and triggered less collections.